File: C#/백준/Silver/9012. 괄호/괄호.cs
using System;

namespace q{
    class w{
        static void Main(){
            int a = int.Parse(Console.ReadLine());
            for (int i =0; i<a; i++){
                int b = 0;
                string c = Console.ReadLine();
                for (int j =0; j<c.Length; j++){
                    if(c[j]=='('){
                        b++;
                    }
                    else{
                        b--;
                        if(b==-1){
                            break;
                        }
                    }
                }
                if(b==0){
                    Console.WriteLine("YES");
                }
                else{
                    Console.WriteLine("NO");
                }
            }
        }
    }
}

File: Python/백준/Silver/9012. 괄호/괄호.py
a=int(input())
for i in range(a):
    b=input(); ck=0
    for j in range(len(b)):
        if(b[j]=='('):
            ck+=1
        else:
            ck-=1
            if(ck==-1):
                break
    if(ck==0):
        print("YES")
    else:
        print("NO")